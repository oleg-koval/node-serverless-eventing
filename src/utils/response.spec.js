const response = require('./response')

describe('respondWith', () => {
    it('with the parser', () => {
        expect(response.respondWith(200, { a: 1 })).toMatchSnapshot()
    })
    it('returns error if wrong status code', () => {
        expect(() => response.respondWith(99, { a: 1 })).toThrowErrorMatchingSnapshot()
    })
    it('w/o data', () => {
        expect(response.respondWith(200)).toMatchSnapshot()
    })
})

describe('send', () => {
    it('should return a response', () => {
        expect(response.send({ a: 1 })).toMatchSnapshot()
    })
})

describe('serverError', () => {
    it('should return a response', () => {
        expect(response.serverError({ a: 1 })).toMatchSnapshot()
    })
})

describe('error', () => {
    it('should return an error response', () => {
        expect(response.error({ a: 1 })).toMatchSnapshot()
    })
    it('should return a validation response', () => {
        expect(response.error({ isJoi: true, details: [{ message: 'test' }] })).toMatchSnapshot()
    })
})
