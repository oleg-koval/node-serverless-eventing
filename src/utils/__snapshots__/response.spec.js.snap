// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error should return a validation response 1`] = `
Object {
  "body": Object {
    "validationErrors": Array [
      Object {
        "message": "test",
      },
    ],
  },
  "headers": Object {},
  "statusCode": 400,
}
`;

exports[`error should return an error response 1`] = `
Object {
  "body": Object {
    "a": 1,
  },
  "headers": Object {},
  "statusCode": 400,
}
`;

exports[`respondWith returns error if wrong status code 1`] = `"status code out of range"`;

exports[`respondWith w/o data 1`] = `
Object {
  "headers": Object {},
  "statusCode": 200,
}
`;

exports[`respondWith with the parser 1`] = `
Object {
  "body": Object {
    "a": 1,
  },
  "headers": Object {},
  "statusCode": 200,
}
`;

exports[`send should return a response 1`] = `
Object {
  "body": Object {
    "a": 1,
  },
  "headers": Object {},
  "statusCode": 200,
}
`;

exports[`serverError should return a response 1`] = `
Object {
  "body": Object {
    "a": 1,
  },
  "headers": Object {},
  "statusCode": 500,
}
`;
